docs/sharing-configurations.md

```diff
@@ -0,0 +1,183 @@
+{
+  "name": "@username/prettier-config",
+  "version": "1.0.0",
+  "description": "My personal Prettier config",
+  "type": "module",
+  "exports": "./index.js",
+  "license": "MIT",
+  "peerDependencies": {
+    "prettier": ">=2.0.0"
+  }
+}
+```
+
+`index.js` is where you put the shared configuration. This file just exports a [regular prettier configuration](./configuration.md) with the same syntax and same options:
+
+```js
+const config = {
+  trailingComma: "es5",
+  tabWidth: 4,
+  singleQuote: true,
+};
+
+export default config;
+```
+
+An example shared configuration repository is available [here](https://github.com/azz/prettier-config).
+
+## Publishing a Shareable Config
+
+Once you are ready, you can [publish your package to npm](https://docs.npmjs.com/creating-and-publishing-scoped-public-packages#publishing-scoped-public-packages):
+
+```sh
+npm publish --access public
+```
+
+## Using a Shareable Config
+
+You first need to install your published configuration, for example:
+
+```sh
+npm install @username/prettier-config --save-dev
+```
+
+Then, you can reference it in your `package.json`:
+
+```json
+{
+  "name": "my-cool-library",
+  "version": "9000.0.1",
+  "prettier": "@username/prettier-config"
+}
+```
+
+If you don’t want to use `package.json`, you can use any of the supported extensions to export a string, e.g. `.prettierrc.json`:
+
+```json
+"@company/prettier-config"
+```
+
+### Extending a Sharable Config
+
+To _extend_ the configuration to overwrite some properties from the shared configuration, import the file in a `.prettierrc.mjs` file and export the modifications, e.g:
+
+```js
+import myPrettierConfig from "@username/prettier-config";
+
+export default {
+  ...myPrettierConfig,
+  semi: false,
+};
+```
+
+## Other examples
+
+### Using Type Annotation in the Shared Config
+
+You can get type safety and autocomplete support in your shared configuration by using a `jsdoc` type annotation:
+
+```js
+/**
+ * @type {import("prettier").Config}
+ */
+const config = {
+  trailingComma: "es5",
+  tabWidth: 4,
+  semi: false,
+  singleQuote: true,
+};
+
+export default config;
+```
+
+In order to make this work, you have to install `prettier` as a `devDependency` for the project:
+
+```sh
+npm install -D prettier
+```
+
+Your `package.json` file should look like this now:
+
+```diff
+{
+  "name": "@username/prettier-config",
+  "version": "1.0.0",
+  "description": "My personal Prettier config",
+  "type": "module",
+  "exports": "./index.js",
+  "license": "MIT",
+  "peerDependencies": {
+    "prettier": ">=2.0.0"
+  },
++ "devDependencies": {
++   "prettier": "^3.3.3"
++ }
+}
+```
+
+### Include Plugins in Shareable Configurations
+
+In case you want to use [plugins](./plugins.md) in your shared configuration, you need to declare those plugins in the config file's `plugin` array and as `dependencies` in `package.json`:
+
+```js
+// index.js
+const config = {
+  singleQuote: true,
+  plugins: ["prettier-plugin-xml"],
+};
```

Comment:
The recommended way to use plugin should be

```js
import * as prettierPlugin ...

const config = {
  singleQuote: true,
  plugins: [prettierPlugin],
};
```

But `vscode-prettier` cannot work since they transfer config to worker.

My suggestion is to use an absolute path/url instead, but `import.meta.resolve` is under `--experimental-import-meta-resolve` flag. Not sure what's the best way here.

Reply:
Fortunately  `import.meta.resolve` by itself is not under a flag, only if someone wants to its second parameter:

![image](https://github.com/user-attachments/assets/d0675af9-2f04-46d3-9522-e19c557439a0)

My issue is now that I've got an error from the `vscode-prettier` extension that it cannot find the module `@prettier/plugin-xml/src/plugin.js`. I'm gonna open a separate issue there.

What do you think about merging this PR without this section about sharing configurations with plugins? It could be added later in a follow up PR.

Reply:
> I'm gonna open a separate issue there.

The issue is here in Prettier.

Reply:
> Fortunately import.meta.resolve by itself is not under a flag,

We still support Node.js v14.

Reply:
> What do you think about merging this PR without this section about sharing configurations with plugins? It could be added later in a follow up PR easily.

The example should work for every one, give me a while to think about it.

Reply:
Sure thing! Take as much time as needed, I'm gonna update the PR when necessary^^

Reply:
Sorry for leaving this for such a long time. Let's merge with current example.